#### List of common packages/libraries 

In [1]:
#supressing the warnings
import warnings
warnings.filterwarnings("ignore")

#data manipulation 
import pandas as pd
import numpy as np

#data vizualization
import matplotlib as plt
import seaborn as sns



#machine learning
#from sklearn import *
#from pycaret import *

##
pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)

#display float to 2 decimal places
pd.options.display.float_format = "{:,.2f}".format

#### reading/importing the data

In [2]:
#### read/importing the data
df = pd.read_csv("../data/AB_NYC_2019.csv")

In [3]:
##checking the dimension of the data
df.shape

(48895, 16)

In [4]:
df.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington     40.65     -73.97   
1     Jennifer           Manhattan       Midtown     40.75     -73.98   
2    Elisabeth           Manhattan        Harlem     40.81     -73.94   
3  LisaRoxanne            Brooklyn  Clinton Hill     40.69     -73.96   
4        Laura           Manhattan   East Harlem     40.80     -73.94   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                nan                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [5]:
df.tail()

id                                               name   host_id  \
48890  36484665    Charming one bedroom - newly renovated rowhouse   8232441   
48891  36485057      Affordable room in Bushwick/East Williamsburg   6570630   
48892  36485431            Sunny Studio at Historical Neighborhood  23492952   
48893  36485609               43rd St. Time Square-cozy single bed  30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
48890        Sabrina            Brooklyn  Bedford-Stuyvesant     40.68   
48891        Marisol            Brooklyn            Bushwick     40.70   
48892  Ilgar & Aysel           Manhattan              Harlem     40.81   
48893            Taz           Manhattan      Hell's Kitchen     40.76   
48894     Christophe           Manhattan      Hell's Kitchen     40.76   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
48890     -73.95     Private room     70               2                  0   
48891     -73.93     Private room     40               4                  0   
48892     -73.95  Entire home/apt    115              10                  0   
48893     -73.99      Shared room     55               1                  0   
48894     -73.99     Private room     90               7                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
48890         NaN                nan                               2   
48891         NaN                nan                               2   
48892         NaN                nan                               1   
48893         NaN                nan                               6   
48894         NaN                nan                               1   

       availability_365  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23

In [6]:
#checking the target variable
df.room_type.value_counts()

Entire home/apt    25409
Private room       22326
Shared room         1160
Name: room_type, dtype: int64

In [8]:
# Function to calculate missing values by columns
def missing_values_table(df):
    #Total missing values 
    mis_val = df.isnull().sum()
    
    #percentage of missing values
    mis_val_percent = 100 * df.isnull().sum() / len(df)
    
    # Make a table with the results
    mis_val_table = pd.concat([mis_val,mis_val_percent], axis =1)
    
    # Rename the colums
    mis_val_table_ren_columns = mis_val_table.rename(
    columns = {0 : 'Missing Values', 1 : '% of Total Values'})
    
    # Sort the table by percentage of missing descending
    mis_val_table_ren_columns = mis_val_table_ren_columns[
        mis_val_table_ren_columns.iloc[:,1] != 0].sort_values(
        '% of Total Values', ascending=False).round(1)
    
    # Print some summary information
    print("Your selected dataframe has " + str(df.shape[1])+ " columns.\n"
         "There are " + str(mis_val_table_ren_columns.shape[0])+
         " columns that have missing values.")
    
    # Return the dataframe with missing information
    return mis_val_table_ren_columns

In [9]:
#checking for missing values
missing_values_table(df)

Your selected dataframe has 16 columns.
There are 4 columns that have missing values.


Missing Values  % of Total Values
last_review                 10052              20.60
reviews_per_month           10052              20.60
host_name                      21               0.00
name                           16               0.00

`last_review` and `reviews_per_month` have ~21% missing values. We shall impute them